src/server.js

```diff
@@ -283,28 +283,35 @@ class Server {
    * @param {Function} handler Handler to perform the work.
    * @param {Object} job Job for the worker to perform.
    * @param {Function} done RabbitMQ acknowledgement callback.
+   * @return {Promise}
+   * @resolves {undefined} when worker is successful
+   * @rejects {Error} when creating Worker fails
    */
   _runWorker (
     queueName: string,
     handler: Function,
     job: Object,
     done: Function
-  ): void {
-    this.log.trace({
-      queue: queueName,
-      job: job,
-      method: '_runWorker'
-    }, 'running worker')
-    const opts = clone(this._workerOptions[queueName])
-    defaults(opts, {
-      queue: queueName,
-      job: job,
-      task: handler,
-      done: done,
-      log: this.log,
-      errorCat: this.errorCat
+  ): Promise<*> {
+    return Promise.try(() => {
+      this.log.trace({
+        queue: queueName,
+        job: job,
+        method: '_runWorker'
+      }, 'running worker')
+      const opts = clone(this._workerOptions[queueName])
+      defaults(opts, {
+        queue: queueName,
+        job: job,
+        task: handler,
+        log: this.log,
+        errorCat: this.errorCat
+      })
+      const worker = Worker.create(opts)
+      return worker.run().finally(() => {
```

Comment:
This seems to be a style argument not a logic. Since it passes standard we should be good?